pytorch/inference/docker/build_artifacts/torchserve-entrypoint.py

```diff
@@ -15,8 +15,29 @@
 import shlex
 import subprocess
 import sys
+import os
+import signal
 
 
+def reap_zombies(signum, frame):
```

Comment:
Could you run a commit to make sure this doesn't break basic functionality?

You can use our command line utility to run tests for a particular image

Steps:
1. Set `autopatch_build` to "False" in https://github.com/aws/deep-learning-containers/blob/master/pytorch/inference/buildspec-2-3-sm.yml#L8
2. Run the command below (assuming your remote is called 'origin')
```
python src/prepare_dlc_dev_environment.py -b pytorch/inference/buildspec-2-3-sm.yml -cp origin 
```
3. When you've confirmed tests run successfully, revert the temp changes with the command below
```
python src/prepare_dlc_dev_environment.py -rb pytorch/inference/buildspec-2-3-sm.yml -cp origin 
```

Reply:
The test passed by failed to download report
```

================== 4 passed, 14 warnings in 233.86s (0:03:53) ==================
--
3799 | Downloading Test reports for image: 669063966089.dkr.ecr.us-west-2.amazonaws.com/pr-pytorch-inference:2.3.0-cpu-py311-ubuntu20.04-sagemaker-pr-4249-autopatch-2024-09-18-16-46-42
3800 | Downloading executions cache from ec2 instance
3801 | Downloading executions cache from ec2 instance
3802 | Cache file wasn't downloaded: [Errno 2] No such file
3803 | Cache file wasn't downloaded: [Errno 2] No such file
3804 | No cache file was created
3805 | No cache file was created
3806 | Terminating Instances for image: 669063966089.dkr.ecr.us-west-2.amazonaws.com/pr-pytorch-inference:2.3.0-cpu-py311-ubuntu20.04-sagemaker-pr-4249-autopatch-2024-09-18-16-46-42
3807 | Destroying ssh Key_pair for image: 669063966089.dkr.ecr.us-west-2.amazonaws.com/pr-pytorch-inference:2.3.0-cpu-py311-ubuntu20.04-sagemaker-pr-4249-autopatch-2024-09-18-16-46-42
3808 | Traceback (most recent call last):
3809 | File "test/testrunner.py", line 569, in <module>
3810 | main()
3811 | File "test/testrunner.py", line 557, in main
3812 | run_sagemaker_local_tests(testing_image_list, pytest_cache_params)
3813 | File "test/testrunner.py", line 74, in run_sagemaker_local_tests
3814 | raise RuntimeError(
3815 | RuntimeError: SageMaker Local tests failed on the following DLCs:
3816 | [
3817 | "669063966089.dkr.ecr.us-west-2.amazonaws.com/pr-pytorch-inference:2.3.0-gpu-py311-cu121-ubuntu20.04-sagemaker-pr-4249-autopatch-2024-09-18-16-46-42"
3818 | ]
3819 |  
3820 | [Container] 2024/09/18 17:19:21.326559 Command did not exit successfully python test/testrunner.py exit status 1
3821 | [Container] 2024/09/18 17:19:21.330468 Phase complete: BUILD State: FAILED


```